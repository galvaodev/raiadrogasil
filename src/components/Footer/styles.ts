import styled, { css } from 'styled-components'
import media from 'styled-media-query'

interface flexProps {
  flex?: boolean
}

export const Wrapper = styled.footer``

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    justify-content: space-between;
    gap: ${theme.grid.gutter};
    margin-top: ${theme.spacings.medium};
  `}

  ${media.greaterThan('medium')`
    flex-wrap: nowrap;
  `}
`
export const Column = styled.div<flexProps>`
  ${({ theme, flex }) => css`
    text-align: center;

    a,
    span {
      display: block;
      color: ${theme.colors.secondary};
      text-decoration: none;
      margin-bottom: ${theme.spacings.xxsmall};
      font-size: ${theme.font.sizes.small};
    }
    a:hover {
      text-decoration: underline;
    }

    ${flex &&
    css`
      flex: 1;
      display: flex;
      padding-left: ${theme.spacings.medium};

      img {
        margin-right: 10px;
      }

      ${media.lessThan('medium')`
        padding-left: 0;
        flex-wrap: wrap;
        justify-content: space-between;
        text-align: center;

        img {
          height: 25px;
        }
      `}
    `}

    ${media.lessThan('medium')`
      img {
        height: 20px;
      }
    `}
  `}
`
export const Copyright = styled.div`
  ${({ theme }) => css`
    width: auto;
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.xsmall};
    text-align: left;
  `}
`
