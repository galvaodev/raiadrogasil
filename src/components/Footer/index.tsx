import Link from 'next/link'
import Heading from 'components/Heading'
import MediaMatch from 'components/MediaMatch'
import * as S from './styles'

const Footer = () => (
  <S.Wrapper>
    <S.Content>
      <MediaMatch flex={1} lessThan="medium">
        <S.Column flex>
          <Link href="/">
            <img src="/img/logo_drogaraia.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_drogasil.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_farmasil.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_univers.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_4bio.png" alt="RaiaDrogasil" />
          </Link>
        </S.Column>
      </MediaMatch>
      <S.Copyright>
        <Heading color="secondary" size="small">
          RD 2017. Todos os direitos reservados
        </Heading>
      </S.Copyright>

      <MediaMatch flex={1} greaterThan="medium">
        <S.Column flex>
          <Link href="/">
            <img src="/img/logo_drogaraia.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_drogasil.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_farmasil.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_univers.png" alt="RaiaDrogasil" />
          </Link>
          <Link href="/">
            <img src="/img/logo_4bio.png" alt="RaiaDrogasil" />
          </Link>
        </S.Column>
      </MediaMatch>
      <S.Column>
        <img src="/img/logo_rd_small.png" alt="RaiaDrogasil" />
      </S.Column>
    </S.Content>
  </S.Wrapper>
)

export default Footer
