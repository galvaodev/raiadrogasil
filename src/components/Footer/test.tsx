import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the heading column name', () => {
    const { container } = renderWithTheme(<Footer />)

    // Contact
    expect(
      screen.getByRole('heading', {
        name: /RD 2017. Todos os direitos reservados/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
