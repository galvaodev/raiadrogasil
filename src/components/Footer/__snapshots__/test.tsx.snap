// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the heading column name 1`] = `
.c4 {
  color: #868686;
  font-size: 1.4rem;
  font-weight: 300;
}

.c4::after {
  width: 3rem;
}

.c1 {
  display: none;
}

.c5 {
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 0.938rem;
  margin-top: 3.2rem;
}

.c2 {
  text-align: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 3.2rem;
}

.c2 a,
.c2 span {
  display: block;
  color: #868686;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.8rem;
  font-size: 1.4rem;
}

.c2 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 img {
  margin-right: 10px;
}

.c6 {
  text-align: center;
}

.c6 a,
.c6 span {
  display: block;
  color: #868686;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.8rem;
  font-size: 1.4rem;
}

.c6 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  width: auto;
  color: #868686;
  font-size: 1.2rem;
  text-align: left;
}

@media (max-width:768px) {
  .c1 {
    display: block;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

@media (min-width:768px) {
  .c5 {
    display: block;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

@media (max-width:768px) {
  .c2 {
    padding-left: 0;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    text-align: center;
  }

  .c2 img {
    height: 25px;
  }
}

@media (max-width:768px) {
  .c2 img {
    height: 20px;
  }
}

@media (max-width:768px) {
  .c6 img {
    height: 20px;
  }
}

<footer
  class=""
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="RaiaDrogasil"
          src="/img/logo_drogaraia.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_drogasil.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_farmasil.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_univers.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_4bio.png"
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <h2
        class="c4"
        color="secondary"
      >
        RD 2017. Todos os direitos reservados
      </h2>
    </div>
    <div
      class="c5"
    >
      <div
        class="c2"
      >
        <img
          alt="RaiaDrogasil"
          src="/img/logo_drogaraia.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_drogasil.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_farmasil.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_univers.png"
        />
        <img
          alt="RaiaDrogasil"
          src="/img/logo_4bio.png"
        />
      </div>
    </div>
    <div
      class="c6"
    >
      <img
        alt="RaiaDrogasil"
        src="/img/logo_rd_small.png"
      />
    </div>
  </div>
</footer>
`
